control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -0,0 +1,270 @@
+/*
```

Comment:
there is a lot of similarity between the channel and the broker here. I'd hope that eventually we solve this